[STEP4-1] 생성된 데이터가 rdb 를 통해 영속성을 가지도록 변경

MyLittleChicken opened the pull request:
### 구현 내용
- 어플리케이션을 통해 생성된 데이터는 mysql 에 저장됩니다.
- 특정 db 에 종속적이지 않은 형태로 구현하고자 인터페이스를 작성했습니다.
- service layer 에 대해 기본적인 transaction 처리가 될 수 있도록 합니다.

MyLittleChicken committed:
chore: PersistDataMysql.md 작성

MyLittleChicken committed:
chore: jdbc, mysql 관련 의존성 추가 및 application.yml datasource 추가

MyLittleChicken committed:
feat: rdb repository 구현

1. RelationDatabaseRepository 구현
2. SqlProvider 와 구현체인 MysqlProvider 구현
3. 기존의 InMemoryRepository 구현체는 제거
4. 엔티티의 Instant 타입을 LocalDateTime 으로 수정

MyLittleChicken committed:
chore: url-shortener.sql 추가

MyLittleChicken committed:
chore: README.md 최신화

1. 실행에 있어 mysql 이 필요하다는 문구 추가
2. datasource 변수 설정하는 내용 추가

MyLittleChicken committed:
feat: service layer 에 누락되었던 Transactional 어노테이션 추가

Hyune-c commented on src/main/java/community/whatever/onembackendjava/entity/ShortenUrlEntity.java:
```diff
@@ -23,7 +23,7 @@ public static ShortenUrlEntity createEntity(
                 originUrl,
                 false,
-                Instant.now(),
+                LocalDateTime.now(),
```
로직이 전혀 달라졌습니다. 의도된 건가요?

MyLittleChicken replied:
@Hyune-c 
의도치 않은 변경사항이네요..
mapping 방식을 명시적으로 표현하는 객체를 통해 로직의 변경 없이 반환될 수 있도록 수정 해야겠습니다.

Hyune-c commented on src/main/java/community/whatever/onembackendjava/infrastructure/MysqlProvider.java:
```diff
@@ -0,0 +1,38 @@
+                "(shorten_url, origin_url, is_deleted, created_at, expired_at) " +
+                "VALUES (?, ?, ?, ?, ?)";
+    }
+}
```
이게 뭔가 했더니 벤더별로 다를 수 있는 native sql 을 표현해놓은 거네요.
- hibernate 를 쓰면 되긴 하지만, 지금은 제약된 상황이니 이렇게 표현하신 것으로 이해 합니다.
- 근데 그렇다고 쳐도 저는 그냥 상수로 관리할 것 같긴 합니다.

이해하고 나니 읽히긴 하는데, 첨에는 무슨 의도인지 이해하기 힘들었습니다.
특히 사용처의 코드를 봤을 때 더 혼란 스러웠습니다.

MyLittleChicken replied:
@Hyune-c 
상수로 관리하는 것을 고려해보겠습니다.
추가로 메서두명에 prefix 등을 추가하여 가독성을 높이는 방향으로 수정하겠습니다.

MyLittleChicken committed:
feat: RowMapper 구현

MyLittleChicken committed:
refactor: 로직 변경에 대한 복원 및 일부 가독성 향상을 위한 수정

1. 생성일시, 수정일시 타입을 Instant 로 복원
2. 매핑 시 명시적으로 구현한 RowMapper 구현체를 이용하도록 수정
3. SqlProvider 메서드에 대한 "get" prefix, "sql" suffix 적용

MyLittleChicken committed:
refactor: SqlProvider 제거 및 sql 을 열거형 상수로 정의

MyLittleChicken committed:
Merge branch 'MyLittleChicken' into feat/persist-data-mysql
